packages/form/core/form-group.ts

```diff
@@ -1,10 +1,26 @@
 import { ControlConfig, FormControl } from './form-control';
 import ShortUniqueId from 'short-unique-id';
+import type { ResolvedField } from '@astro-reactive/common';
+
+/**
+ * Generate `ControlConfig`s with typed name property.
+ */
+type TypedControlName<T> = Omit<ControlConfig, 'name'> & {
+	name: Extract<keyof T, string>;
+} & ControlConfig;
+
+/**
+ * FormGroup configuration
+ */
+interface FormConfig {
+	name?: string;
+	validationSchema?: ResolvedField;
+}
 
 /**
  *  Represents a group of controls that will be rendered as a fieldset element in a form.
  */
-export class FormGroup {
+export class FormGroup<FormValues = Record<string, never>> {
```

Comment:
Minor suggestion, I think it's good here to use a generic `T` instead of `FormValues` 🤔 Somehow `FormValues` seem to me like it refers to values of form control...